sympy/sets/conditionset.py

```diff
@@ -34,6 +34,8 @@ def __new__(cls, sym, condition, base_set):
             return S.EmptySet
         if condition == S.true:
             return base_set
+        if getattr(base_set, "is_FiniteSet", False) and base_set.is_FiniteSet:
```

Comment:
I think, `getattr` is better. Because, if we pass an `EmptySet` instead of `FiniteSet`, then it won't return an `EmptySet`.

```
In [3]: ConditionSet(x, And(x < -3), EmptySet())
Out[3]: ConditionSet(x, x < -3, EmptySet())
```